File: VB/WpfApplication1/ThemeHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Xpf.Ribbon
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.Bars
Imports System.Reflection
Imports System.Windows.Media.Imaging
Imports System.Windows.Media

Namespace ThemeHelperWPF

    Public Module ThemeHelper

        Public Sub InitThemeSubItems(ByVal barSubTheme As BarSubItem)
            ThemeHelper.InitThemeSubItems(barSubTheme, True)
        End Sub

        Public Sub InitThemeSubItems(ByVal barSubTheme As BarSubItem, ByVal Optional _UseLargeIcon As Boolean = True)
            For Each theme As Theme In Theme.Themes
                Dim item As BarButtonItem = New BarButtonItem() With {.Content = theme.Name, .Glyph = LoadImageByThemeName(theme.Name, _UseLargeIcon)}
                item.ItemClick += New ItemClickEventHandler(AddressOf item_ItemClick)
                barSubTheme.ItemLinks.Add(item)
            Next
        End Sub

        Public Sub InitThemeGallery(ByVal galleryBarItem As RibbonGalleryBarItem)
            ThemeHelper.InitThemeGallery(galleryBarItem, True)
        End Sub

        Public Sub InitThemeGallery(ByVal galleryBarItem As RibbonGalleryBarItem, ByVal Optional _UseLargeIcon As Boolean = True)
            InitThemeGallery(galleryBarItem.Gallery, _UseLargeIcon)
        End Sub

        Public Sub InitThemeGallery(ByVal gallery As Gallery)
            ThemeHelper.InitThemeGallery(gallery, True)
        End Sub

        Public Sub InitThemeGallery(ByVal gallery As Gallery, ByVal Optional _UseLargeIcon As Boolean = True)
            gallery.ItemClick += New GalleryItemEventHandler(AddressOf Gallery_ItemClick)
            For Each theme As Theme In Theme.Themes
                gallery.Groups(0).Items.Add(New GalleryItem() With {.Caption = theme.Name, .Glyph = LoadImageByThemeName(theme.Name, _UseLargeIcon)})
            Next
        End Sub

        Private Sub Gallery_ItemClick(ByVal sender As Object, ByVal e As GalleryItemEventArgs)
            ThemeManager.ApplicationThemeName = e.Item.Caption.ToString()
        End Sub

        Private Sub item_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            ThemeManager.ApplicationThemeName = e.Item.Content.ToString()
        End Sub

        Private Function LoadImageByThemeName(ByVal name As String, ByVal _UseLargeIcon As Boolean) As ImageSource
            Dim fileName As String = Nothing
            If _UseLargeIcon Then
                fileName = name
            Else
                fileName = name & "_24x24"
            End If

            Return New BitmapImage(New Uri("pack://application:,,,/" & Assembly.GetExecutingAssembly().GetName().Name & ";component/" & "Images/" & fileName & ".png", UriKind.Absolute))
        End Function
    End Module
End Namespace

File: CS/WpfApplication1/ThemeHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpf.Ribbon;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.Bars;
using System.Reflection;
using System.Windows.Media.Imaging;
using System.Windows.Media;

namespace ThemeHelperWPF
{
    public static class ThemeHelper
    {
        public static void InitThemeSubItems(BarSubItem barSubTheme) 
        {
            InitThemeSubItems(barSubTheme, true);
        }
        public static void InitThemeSubItems(BarSubItem barSubTheme, bool _UseLargeIcon=true)
        {
            foreach (Theme theme in Theme.Themes)
            {
                if (theme.Name == "HybridApp") break;
                BarButtonItem item = new BarButtonItem()
                {
                    Content = theme.Name,
                    Glyph = LoadImageByThemeName(theme.Name, _UseLargeIcon),
                };
                item.ItemClick += new ItemClickEventHandler(item_ItemClick);
                barSubTheme.ItemLinks.Add(item);
            }
        }
        public static void InitThemeGallery(RibbonGalleryBarItem galleryBarItem)
        {
            InitThemeGallery(galleryBarItem, true);
        }
        public static void InitThemeGallery(RibbonGalleryBarItem galleryBarItem, bool _UseLargeIcon=true)
        {
            InitThemeGallery(galleryBarItem.Gallery, _UseLargeIcon);
        }
        public static void InitThemeGallery(Gallery gallery)
        {
            InitThemeGallery(gallery, true);
        }
        public static void InitThemeGallery(Gallery gallery, bool _UseLargeIcon = true)
        {
            gallery.ItemClick += new GalleryItemEventHandler(Gallery_ItemClick);
            foreach (Theme theme in Theme.Themes)
            {
                if (theme.Name == "HybridApp") break;
                gallery.Groups[0].Items.Add(new GalleryItem()
                {
                    Caption = theme.Name,
                    Glyph = LoadImageByThemeName(theme.Name, _UseLargeIcon)
                });
            }
        }
        static void Gallery_ItemClick(object sender, GalleryItemEventArgs e)
        {
            ThemeManager.ApplicationThemeName = e.Item.Caption.ToString();
        }
        static void item_ItemClick(object sender, ItemClickEventArgs e)
        {
            ThemeManager.ApplicationThemeName = e.Item.Content.ToString();
        }
        private static ImageSource LoadImageByThemeName(string name, bool _UseLargeIcon)
        {
            string fileName = null;
            if (_UseLargeIcon) 
            {
                fileName = name + "_48x48";
            }
            else
            {
                fileName = name + "_16x16";
            }
            return new BitmapImage(new Uri(@"pack://application:,,,/"
             + Assembly.GetExecutingAssembly().GetName().Name
             + ";component/"
             + "Images/" + fileName + ".png", UriKind.Absolute));
        }
    }
}
